File: 19주차/이연상/영어 끝말잇기.py
# [PRG] 영어 끝말잇기

import math

def solution(n, words):
    temp = []
    for i in range(len(words)):
        if not temp:
            temp.append(words[i])
            continue
        if (words[i] not in temp):
            s = temp[-1]
            now = words[i]
            if(s[-1] == now[0]):
                temp.append(words[i])
                continue
        
        
        if (i + 1) % n == 0: b = n
        else : b = (i + 1) % n
        a = math.ceil((i + 1) / n)
        return [b, a]
        

    return [0, 0]

File: 43주차/이연상/영어 끝말잇기.java
// [PRG] 영어 끝말잇기

import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];

        ArrayList<String> wordList = new ArrayList<>();
        for(int i = 0; i < words.length; i++) {
            if(i == 0) {
                wordList.add(words[i]);
                continue;
            }
            String temp = wordList.get(wordList.size() - 1);
            if(wordList.contains(words[i]) || 
               temp.charAt(temp.length() - 1) != words[i].charAt(0)) {
                if((i + 1) % n == 0) {
                    answer[0] = n;
                    answer[1] = (i + 1) / n;
                }
                else {
                    answer[0] = (i + 1) % n;
                    answer[1] = (i + 1 + n) / n;
                }  
                break;
            }
            wordList.add(words[i]);
        }
        
        return answer;
    }
}